### PR TITLE
First Draft EyeLike CPU OP9

### PR DESCRIPTION
**Pull request by askhade**
Adding implementation for new op-9 operator eyelike

### CODE REVIEW

File: onnxruntime/core/providers/cpu/tensor/eye_like.h [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+#include "core/common/common.h"
+#include "core/framework/op_kernel.h"
+
+namespace onnxruntime {
+
+class EyeLike final : public OpKernel {
+ public:
+  EyeLike(const OpKernelInfo& info) : OpKernel(info) {
+    if (!info.GetAttr("k", &k_).IsOK()) {
+      ONNXRUNTIME_THROW("Missing 'k' attribute value");
```
Comment (pranavsharma) [inline onnxruntime/core/providers/cpu/tensor/eye_like.h]: Looks like 'k' is optional based on the spec in which case we don't have to throw.

File: onnxruntime/core/providers/cpu/tensor/eye_like.cc [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+  if (input_dims.size() != 2) {
+    return Status(ONNXRUNTIME, INVALID_ARGUMENT, "EyeLike : Input tensor dimension is not 2");
+  }
+
+  // set output tensor shape same as input tensor and set all values to zero
+  auto* T2 = context->Output(0, input_dims);
+  auto* data = T2->MutableData<T>();
+  T zero_value = static_cast<T>(0);
+  auto out = gsl::make_span(data, T2->Shape().Size());
+  std::for_each(out.begin(), out.end(), [&zero_value](T& v) { v = zero_value; });
```
Comment (pranavsharma) [inline onnxruntime/core/providers/cpu/tensor/eye_like.cc]: Consider using std::fill which makes the intent obvious.

File: onnxruntime/core/providers/cpu/tensor/eye_like.cc [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+using namespace ::onnxruntime::common;
+
+namespace onnxruntime {
+
+ONNX_CPU_OPERATOR_KERNEL(
+    EyeLike,
+    9,
+    KernelDefBuilder().TypeConstraint("T1",
+                                      std::vector<MLDataType>{
+                                          DataTypeImpl::GetTensorType<float>(),
```
[the remark this answers is not in the document]
Comment (souptc) [reply onnxruntime/core/providers/cpu/tensor/eye_like.cc]: we could add the types step by step based on requirement. Is there any models require other types?

[the remark this answers is not in the document]
Comment (askhade) [reply onnxruntime/core/providers/cpu/tensor/eye_like.cc]: I don't know if any model actually needs this operator right now. I am adding this as part of OP9 support. Do we maintain this info somewhere? How do I check?

[the remark this answers is not in the document]
Comment (duli2012) [reply onnxruntime/core/providers/cpu/tensor/eye_like.cc]: By default, we start with tensor(float).

File: onnxruntime/core/providers/cpu/tensor/eye_like.cc [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+
+  // set output tensor shape same as input tensor and set all values to zero
+  auto* T2 = context->Output(0, input_dims);
+  auto output_mat = EigenMatrixMapRowMajor<T>(
+      T2->template MutableData<T>(),
+      input_dims[0],
+      input_dims[1]);
+  output_mat.setZero();
+  
+  if ((k_ >= 0 && k_ >= input_dims[1]) || (k_ < 0 && std::abs(k_) >= input_dims[0])) {
```
Comment (pranavsharma) [inline onnxruntime/core/providers/cpu/tensor/eye_like.cc]: is this an error condition?

**Pull request merged**